src/app/account/invitation/[inviteId]/invitation-handler.tsx

```diff
@@ -0,0 +1,128 @@
+import { useDashboardUrl } from '@/lib/dashboard-url'
+
+interface InvitationHandlerProps {
+    inviteId: string
+    invitedEmail: string
+}
+
+export function InvitationHandler({ inviteId, invitedEmail }: InvitationHandlerProps) {
+    const { isLoaded, isSignedIn } = useAuth()
+    const { setActive } = useClerk()
+    const { user } = useUser()
+    const router = useRouter()
+    const [isLoadingAction, setIsLoadingAction] = useState(false)
+    const [newOrgSlug, setNewOrgSlug] = useState<string | null>(null)
+    const [orgName, setOrgName] = useState<string | null>(null)
+    const [error, setError] = useState<string | null>(null)
+    // use dynamic dashboard resolution
+    const dashboardUrl = useDashboardUrl()
+
+    // remember the inviteId so that once they sign in we auto-claim
+    useEffect(() => {
+        window.localStorage.setItem('pendingInviteId', inviteId)
+    }, [inviteId])
+
+    useEffect(() => {
+        if (isLoaded && isSignedIn) {
+            setIsLoadingAction(true)
+            claimInviteAction({ inviteId })
+                .then((result) => {
+                    if (result.success && result.organizationName) {
+                        setOrgName(result.organizationName)
+                        setNewOrgSlug(result.orgSlug || null)
+                        // The success message is now part of the SuccessPanel
+                    } else {
+                        setError(result.error || 'Failed to accept invitation.')
+                        notifications.show({
+                            title: 'Error',
+                            message: result.error || 'Could not process your invitation.',
+                            color: 'red',
+                        })
+                    }
+                })
+                .catch((err) => {
+                    reportError(err, 'Error accepting invitation for existing user')
+                    setError('An unexpected error occurred while processing your invitation.')
+                    notifications.show({
+                        title: 'Error',
+                        message: 'An unexpected error occurred.',
+                        color: 'red',
+                    })
+                })
+                .finally(() => setIsLoadingAction(false))
+        }
+    }, [isLoaded, isSignedIn, inviteId])
+
+    useEffect(() => {
+        if (!newOrgSlug || !user) return
+
+        const orgMeta = user.publicMetadata?.orgs?.find((o) => o.slug === newOrgSlug)
+        if (orgMeta?.isReviewer) {
+            // update Clerk’s active organization so useDashboardUrl picks it up
+            setActive({ organization: newOrgSlug }).catch(() => {})
+        }
+    }, [newOrgSlug, user, setActive])
+
+    // fetch whether this email is already in our SI user table
+    const { data: userExists } = useQuery({
+        queryKey: ['userExists', invitedEmail],
+        queryFn: () => userExistsAction(invitedEmail),
+        enabled: isLoaded,
+    })
```

Comment:
can this all be done as part of the `claimInviteAction`  and return error if it's already in use?